### PR TITLE
cdc/sink: mysql sink manage checkpoint per table

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve? 
To fix the bug of #3503 , we should manage table checkpoint per table in the sink, this is PR is a part of the fix. In this PR, mysql sink manage checkpointTs per table

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: tests/integration_tests/sink_hang/run.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ function run() {
 	*) SINK_URI="mysql://normal:123456@127.0.0.1:3306/?max-txn-row=1" ;;
 	esac
 
-	export GO_FAILPOINTS='github.com/pingcap/ticdc/cdc/sink/MySQLSinkHangLongTime=1*return(true);github.com/pingcap/ticdc/cdc/sink/MySQLSinkExecDMLError=9*return(true)'
+	export GO_FAILPOINTS='github.com/pingcap/ticdc/cdc/sink/MySQLSinkExecDMLError=2*return(true)'
```
Comment (asddongmen) [inline tests/integration_tests/sink_hang/run.sh:45]: Can the same test purpose be achieved after changed this line?

**Review by asddongmen — APPROVED** [review]

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**